src/model/table.py

```diff
@@ -1,15 +1,35 @@
+import os
+
 from model.table_metadata import *
 
 
 class Table:
-    def __init__(self, database, table):
-        self.__path__ = os.path.join(database.get_path(), table[Keys.NAME])
-        self.__table_metadata__ = TableMetaData(table, self)
+    def __init__(self, database, table=None, table_name=None):
+        if table is not None:
+            self.__initialize_by_table_schema__(database, table)
+        elif table_name is not None:
+            self.__initialize_by_table_name__(database, table_name)
+        else:
+            raise NoParameterError("No table detected")
 
     def serialize(self):
         os.makedirs(self.__path__, exist_ok=True)
         self.__table_metadata__.serialize()
 
+    def __initialize_by_table_schema__(self, database, table_schema):
```

Comment:
You don’t need to have 2 init here. It can be one init.
Think about it.
 Also make the parameter to init meaningful by replacing table by table_name